src/Smuggler/Anns.hs

```diff
@@ -14,3 +18,33 @@ removeAnnAtLoc line col = Map.filterWithKey (\k _ -> matchKey k)
     matchKey (AnnKey (RealSrcSpan rss) _) =
         not (srcSpanStartLine rss == line && srcSpanStartCol rss == col)
     matchKey _ = True
+
+removeTrailingCommas :: Anns -> Anns
+removeTrailingCommas
+  = Map.fromList . concatMap removeIfImportDecl
+  . groupBy withinSrcSpan . Map.toList
+  where
+    removeIfImportDecl :: [(AnnKey, Annotation)] -> [(AnnKey, Annotation)]
+    removeIfImportDecl gAnns
+      | any isImportDecl gAnns = removeTrailingComma gAnns
+      | otherwise = gAnns
+
+    removeTrailingComma :: [(AnnKey, Annotation)] -> [(AnnKey, Annotation)]
```

Comment:
Can you please add comments to this function explaining this special case with three elements in the list?

Reply:
Will do, though I'm not quite sure if I can articulate it properly :sweat_smile: